docs/docs/topics/graphql.mdx

```diff
@@ -15,31 +15,33 @@ If you need to extract the current GraphQL schema in your environment you can is
 
 ## Query & mutations
 
-For each model in the schema, a GraphQL query and 3 mutations will be generated based on the namespace and the name of the model.
+In GraphQL, a query is used to fetch data and mutations are use to create/update or delete data. In Infrahub, a GraphQL query and 4 mutations will be generated for each model you define in the schema. The name of the query or mutation is based on the namespace and name of the model.
 
 For example, for the model `CoreRepository` the following query and mutations have been generated:
 
-- `Query` : **CoreRepository**
-- `Mutation` : **CoreRepositoryCreate**
-- `Mutation` : **CoreRepositoryUpdate**
-- `Mutation` : **CoreRepositoryDelete**
+- `Query` : **CoreRepository** to fetch `CoreRepository` nodes from Infrahub
+- `Mutation` : **CoreRepositoryCreate** to create a `CoreRepository` node
+- `Mutation` : **CoreRepositoryUpdate** to update an existing `CoreRepository` node
+- `Mutation` : **CoreRepositoryUpsert** to create or update a `CoreRepository` node
+- `Mutation` : **CoreRepositoryDelete** to delete a `CoreRepository` node
 
 ### Query format
 
 The top level query for each model will always return a list of objects and the query will have the following format `CoreRepository` > `edges` > `node` > `display_label`
 
 ```graphql
 query {
-    CoreRepository {            # PaginatedCoreRepository object
-        count
-        edges {                 # EdgedCoreRepository object
-            node {              # CoreRepository object
-                id
-                display_label
-                __typename
-            }
+  CoreRepository {            # PaginatedCoreRepository object
+    count
+      edges {                 # EdgedCoreRepository object
+        node {                # CoreRepository object
+          id
+          hfid
+          display_label
+          __typename
```

Comment:
```suggestion
  CoreRepository {            # PaginatedCoreRepository object
    count
    edges {                 # EdgedCoreRepository object
      node {                # CoreRepository object
        id
        hfid
        display_label
        __typename
```

Reply:
You do not need to use the brackets, as long as the hfid is made up of a single field. If the hfid is made up of multiple fields, you HAVE to use brackets.

My personal opinion is that we should keep things simple and only have one way of doing things and therefor I prefer to always use the brackets.

Not sure if there's other opinions on that?